### PR TITLE
feat: improve lens display logic

### PR DESCRIPTION
**Pull request by nuanyang233**
## Description

Closes #MF-4028 MF-4033

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/shared/src/UI/components/SocialAccountList/SocialListItem.tsx [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+    const { classes } = useStyles()
+    const { value: address } = useAsync(() => {
+        return ENS.lookup(domain)
+    }, [domain])
+
+    if (!address) return null
+
+    return (
+        <div className={classes.address}>
+            <div className={classes.addressText}>{address}</div>
```
Comment (guanbinrui) [inline packages/shared/src/UI/components/SocialAccountList/SocialListItem.tsx]: `<Typography />`

File: packages/shared/src/UI/components/SocialAccountList/SocialListItem.tsx [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+    return (
+        <SocialTooltip platform={platform}>
+            <MenuItem
+                className={classes.listItem}
+                disableRipple
+                disabled={false}
+                onClick={() => {
+                    if (platform === NextIDPlatform.ENS) {
+                        ENS.lookup(identity).then((address) => {
```
Comment (guanbinrui) [inline packages/shared/src/UI/components/SocialAccountList/SocialListItem.tsx]:
```suggestion
                        const address = await ENS.lookup(identity)
```

File: packages/plugins/Web3Profile/src/locales/en-US.json [hunk 3]
```diff
@@ -9,6 +9,8 @@
     "NFTs": "NFTs",
     "web3_profile": "Web3 Profile",
     "EVM_wallet": "EVM Wallet",
+    "edit_profile_in_lenster": "Edit your profile in lenster",
+    "edit_profile_tips": "{{profile}} is owned by the current connected wallet. Please go to lenster.xyz to edit profile.",
```
Comment (UncleBill) [inline packages/plugins/Web3Profile/src/locales/en-US.json]: Might contain characters that will be escaped.
```suggestion
    "edit_profile_tips": "{{- profile}} is owned by the current connected wallet. Please go to lenster.xyz to edit profile.",
```

File: packages/shared/src/UI/components/SocialAccountList/SocialListItem.tsx [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+
+    const { loading, value } = useAsync(async () => {
+        if (platform !== NextIDPlatform.LENS || !identity) return
+        const profile = await Lens.getProfileByHandle(identity)
+        const isFollowing = await Lens.queryFollowStatus(account, profile.id)
+        return {
+            ownedBy: profile.ownedBy,
+            isFollowing,
+        }
+    }, [identity, platform, account])
```
Comment (UncleBill) [inline packages/shared/src/UI/components/SocialAccountList/SocialListItem.tsx]: Critical dependency.
```suggestion
    }, [identity, platform !== NextIDPlatform.LENS, account])
```

Or
```ts
const isLens = platform === NextIDPlatform.LENS
```

```suggestion
    }, [identity, isLens, account])
```

File: packages/shared/src/UI/components/SocialAccountList/SocialListItem.tsx [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+
+    return (
+        <SocialTooltip platform={platform}>
+            <MenuItem
+                className={classes.listItem}
+                disableRipple
+                disabled={false}
+                onClick={async () => {
+                    if (platform === NextIDPlatform.ENS) {
```
Comment (UncleBill) [inline packages/shared/src/UI/components/SocialAccountList/SocialListItem.tsx:168]: Aha, we can reuse `isLens` here.

File: packages/plugins/Web3Profile/src/SNSAdaptor/components/FollowLensDialog.tsx [hunk 6]
```diff
@@ -315,16 +317,31 @@ export function FollowLensDialog() {
...
-                                endIcon={<Icons.LinkOut size={18} />}
-                                sx={{ cursor: 'pointer' }}>
-                                {t.lenster()}
-                            </Button>
+                            {isSelf ? (
+                                <Button
+                                    variant="roundedContained"
+                                    className={classes.followAction}
+                                    href={profile?.handle ? getLensterLink(profile.handle) : '#'}
+                                    target="__blank"
```
Comment (UncleBill) [inline packages/plugins/Web3Profile/src/SNSAdaptor/components/FollowLensDialog.tsx]:
```suggestion
                                    target="_blank"
```

File: packages/plugins/Web3Profile/src/SNSAdaptor/components/FollowLensDialog.tsx [hunk 7]
```diff
@@ -315,16 +317,31 @@ export function FollowLensDialog() {
...
+                                    sx={{ cursor: 'pointer' }}>
+                                    {t.edit_profile_in_lenster()}
+                                </Button>
+                            ) : (
+                                <>
+                                    {element}
+                                    <Button
+                                        variant="roundedOutlined"
+                                        href={profile?.handle ? getLensterLink(profile.handle) : '#'}
+                                        target="__blank"
```
Comment (UncleBill) [inline packages/plugins/Web3Profile/src/SNSAdaptor/components/FollowLensDialog.tsx]: ditto

File: packages/shared/src/UI/components/SocialAccountList/SocialListItem.tsx (hunk 5 above)
Comment (nuanyang233) [reply packages/shared/src/UI/components/SocialAccountList/SocialListItem.tsx:168]: Here is't `NextIdPlatform.Lens`

File: packages/shared/src/UI/components/SocialAccountList/SocialListItem.tsx (hunk 4 above)
Comment (nuanyang233) [reply packages/shared/src/UI/components/SocialAccountList/SocialListItem.tsx]: There are other platform judgments in this component, such as ens, so I used the first example

**Review by UncleBill — APPROVED** [review]

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**